Src/Web/Web.Shared.Net/PostSamplingTelemetryProcessor.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace Microsoft.ApplicationInsights.Web
+{
+    using System;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    /// <summary>
+    /// PostSamplingTelemetry processor which ensures that the RequestTelemetry.Url is initialized
+    /// in the context of HttpContext.Current.Request
+    /// </summary>
+    public class PostSamplingTelemetryProcessor : ITelemetryProcessor
```

Comment:
lets comment here to indicate this is not to be used by customers, unless they know the implications.